lib/index.js

```diff
@@ -424,61 +424,79 @@ ABI.stringify = function (types, values) {
   return ret
 }
 
+ABI.solidityHexValue = function (type, value, bitsize) {
+  // pass in bitsize = null if use default bitsize
+  var size, num
+  if (type.match('^[a-zA-Z0-9_]+(\\[[0-9]*\\])+$')) {
```

Comment:
There is a function 'isArray' here. Not a regex, but a check for ']' in the end. It ain't much, but it's honest work. Should be enough.